### PR TITLE
FR-1728: BSP - Warning if addressline1 or postcode missing

### PR DESCRIPTION
**Pull request by johnoneill46**
https://tools.hmcts.net/jira/browse/FR-1728

On the Form A paper form - there is only a single text area where users populate the address - this is for both the applicant OR their solicitor (and same for respondent/resp solicitor).

We have separate logic to map the addresses properly based on if they are represented or not.

In this PR we just want to verify that address line 1 and postcode and given for the applicant and respondent (or their respective solicitors) and so I have made the 4 following fields mandatory:
- "ApplicantAddressLine1"
- "ApplicantAddressPostcode"
- "RespondentAddressLine1"
- "RespondentAddressPostcode"

### CODE REVIEW

File: src/functionalTests/resources/json/bulkscan/formA.json [hunk 1]
```diff
@@ -201,12 +201,6 @@
     { "name": "ApplicantRepresented", "value": "I am not represented by a solicitor in these proceedings" },
     { "name": "ApplicantSolicitorName", "value": "Saul Call" },
     { "name": "ApplicantSolicitorFirm", "value": "Better Divorce Ltd" },
-    { "name": "ApplicantSolicitorAddressLine1", "value": "20 Solicitors Road" },
-    { "name": "ApplicantSolicitorAddressLine2", "value": "somewhere" },
-    { "name": "ApplicantSolicitorAddressTown", "value": "Soltown" },
-    { "name": "ApplicantSolicitorAddressCounty", "value": "East Midlands" },
-    { "name": "ApplicantSolicitorAddressPostcode", "value": "GL51 0EX" },
-    { "name": "ApplicantSolicitorAddressCountry", "value": "UK" },
```
Comment (johnoneill46) [inline src/functionalTests/resources/json/bulkscan/formA.json:209]: Removing these values from test data as they are never actually captured from Exela

File: src/functionalTests/resources/json/bulkscan/formA.json [hunk 2]
```diff
@@ -226,12 +220,6 @@
     { "name": "RespondentAddressCounty", "value": "Here" },
     { "name": "RespondentAddressPostcode", "value": "SW1 9SD" },
     { "name": "RespondentAddressCountry", "value": "UK" },
-    { "name": "RespondentSolicitorAddressLine1", "value": "Drive" },
-    { "name": "RespondentSolicitorAddressLine2", "value": "Road" },
-    { "name": "RespondentSolicitorAddressTown", "value": "Leeds" },
-    { "name": "RespondentSolicitorAddressCounty", "value": "Where" },
-    { "name": "RespondentSolicitorAddressPostcode", "value": "SW9 USB" },
-    { "name": "RespondentSolicitorAddressCountry", "value": "Scotland" },
```
Comment (johnoneill46) [inline src/functionalTests/resources/json/bulkscan/formA.json:234]: Removing these values from test data as they are never actually captured from Exela

File: src/functionalTests/resources/json/bulkscan/formA.json [hunk 3]
```diff
@@ -201,12 +201,6 @@
     { "name": "ApplicantRepresented", "value": "I am not represented by a solicitor in these proceedings" },
     { "name": "ApplicantSolicitorName", "value": "Saul Call" },
     { "name": "ApplicantSolicitorFirm", "value": "Better Divorce Ltd" },
-    { "name": "ApplicantSolicitorAddressLine1", "value": "20 Solicitors Road" },
```
Comment (jvictorcf) [inline src/functionalTests/resources/json/bulkscan/formA.json:204]: Shouldn't you have added the new fields to this file? i.e. Applicant's and respondent's line 1 and postcode.

Comment (jvictorcf) [reply src/functionalTests/resources/json/bulkscan/formA.json:204]: And if you didn't, shouldn't your tests be failing?

Comment (johnoneill46) [reply src/functionalTests/resources/json/bulkscan/formA.json:204]: The fields already exist "ApplicantAddressLine1", "ApplicantAddressPostcode", "RespondentAddressLine1" & "RespondentAddressPostcode" in this test data file. The ones I removed were just redundant test data that was never used anywhere (unsure why they were added in the first place)

Comment (jvictorcf) [reply src/functionalTests/resources/json/bulkscan/formA.json:204]: Got you. The ones you're removing are regarding the solicitors, not the citizens.
If that's what you want, I'm happy with it.

**Pull request merged**